--- NaServer.rb.orig	2014-04-23 05:07:01.984514600 -0500
+++ NaServer.rb	2014-04-23 09:09:08.030093900 -0500
@@ -18,7 +18,7 @@
 include REXML
 require 'stringio'
 include StreamListener
-require 'NaElement'
+require File.dirname(__FILE__) + "/NaElement"
 require 'pathname'
 
 # Official NMSDK release version
@@ -572,8 +572,14 @@
     i = i + 1
     value = args[i]
     i = i + 1
+      if value.class == NaElement
+        x = NaElement.new(key)
+        x.child_add(value)
+        xi.child_add(x)
+      else
     xi.child_add(NaElement.new(key, value))
     end
+    end
     return invoke_elem(xi)
   end
 
